File: src/Microsoft.Management.Configuration.Processor.Dsc/GetAllSettingsResult.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
#include "pch.h"
#include "GetAllSettingsResult.h"

namespace winrt::Microsoft::Management::Configuration::Processor::Dsc::implementation
{
    using namespace winrt::Microsoft::Management::Configuration;
    using namespace Windows::Foundation::Collections;

    GetAllSettingsResult::GetAllSettingsResult(const ConfigurationUnit& unit)
        : m_unit(unit) {}

    ConfigurationUnit GetAllSettingsResult::Unit()
    {
        return m_unit;
    }

    IVector<ValueSet> GetAllSettingsResult::Settings()
    {
        return m_settings;
    }

    IConfigurationUnitResultInformation GetAllSettingsResult::ResultInformation()
    {
        return m_resultInformation;
    }

    void GetAllSettingsResult::ResultInformation(IConfigurationUnitResultInformation resultInformation)
    {
        m_resultInformation = resultInformation;
    }
}

File: src/Microsoft.Management.Configuration.Processor/Unit/GetAllSettingsResult.cs
// -----------------------------------------------------------------------------
// <copyright file="GetAllSettingsResult.cs" company="Microsoft Corporation">
//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
// </copyright>
// -----------------------------------------------------------------------------

namespace Microsoft.Management.Configuration.Processor.Unit
{
    using System.Collections.Generic;
    using Microsoft.Management.Configuration;
    using Windows.Foundation.Collections;

    /// <summary>
    /// Implements IGetAllSettingsResult.
    /// </summary>
    internal partial class GetAllSettingsResult : IGetAllSettingsResult
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="GetAllSettingsResult"/> class.
        /// </summary>
        /// <param name="unit">The configuration unit that the result is for.</param>
        public GetAllSettingsResult(ConfigurationUnit unit)
        {
            this.Unit = unit;
        }

        /// <summary>
        /// Gets the configuration unit that the result is for.
        /// </summary>
        public ConfigurationUnit Unit { get; private set; }

        /// <inheritdoc/>
        public IConfigurationUnitResultInformation ResultInformation
        {
            get { return this.InternalResult; }
        }

        /// <summary>
        /// Gets the implementation object for ResultInformation.
        /// </summary>
        public ConfigurationUnitResultInformation InternalResult { get; } = new ConfigurationUnitResultInformation();

        /// <inheritdoc/>
        public IList<ValueSet>? Settings { get; internal set; }
    }
}
